plugins/modules/aci_rest.py

```diff
@@ -413,12 +414,20 @@ def main():
     aci.method = aci.params.get('method').upper()
 
     # Perform request
-    resp, info = fetch_url(module, aci.url,
-                           data=payload,
-                           headers=aci.headers,
-                           method=aci.method,
-                           timeout=aci.params.get('timeout'),
-                           use_proxy=aci.params.get('use_proxy'))
+    resp = None
+    if module._socket_path:
+        connect = Connection(aci.module._socket_path)
+        connect.set_params(aci.headers.get('Cookie'), aci.params)
+        info = connect.send_request(aci.method, '/{0}'.format(path), payload)
+        aci.url = info.get('url')
+        aci.httpapi_logs.extend(connect.pop_messages())
+    else:
+        resp, info = fetch_url(module, aci.url,
+                               data=payload,
+                               headers=aci.headers,
+                               method=aci.method,
+                               timeout=aci.params.get('timeout'),
+                               use_proxy=aci.params.get('use_proxy'))
```

Comment:
Why not to use the request() or api_call() method here too?

Reply:
I added the changes here when api_call() was not successful when it was called from here.